### ISSUE TITLE
Annotation rule small change to annotation intersection rule

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-site#1110]
At present the file I create allows exceptions for GO terms and for specific genes.

I would like to migrate the specific gene IDs to panther family IDs

So- in the "excluded gene" column in the next update I will also include panther family IDs in the form
Panther:PTHR12234 (in fact I will probably change all of the existing gene  ones)

Also where doe shte file live? I can't remember?

**Comment by kltm** [comment]
@ValWood The rules exist here: https://github.com/geneontology/shared-annotation-check/blob/master/rules.txt

For a change like you're describing, we'd either want to add a new column for the PANTHER IDs (of the form `PANTHER:PTHR1234`) or use the current file layout, but either way the runner will have to be changed to support such a difference.

**Comment by ValWood** [comment]
I can put these in the same column as the gene specific exceptions if that's OK. It's the only practical way to do families. Otherwise I would need to put the ID of every species that is likely to be annotated.

We still need to be able to make gene specific exceptions for tandem fusions.

Can the code be made to parse the Panter family ID's because the next version of the rules will necessarily have lots of these.

I found the rules, but I am not convinced they are reported in the organism specific log files? At least I don't see any in the PomBase file, and I know we have some....

**Comment by kltm** [comment]
@ValWood currently, these error check exist as a separate entity from the rest of the "line-by-line" GO Rules and are not reported. The GO Rule in this case is merely being used as a pointer to the shared-annotation-check reports as documentation. In the future, we'd hope to have these merged better, but their history as a separate entity is holding that back a little here.

**Comment by kltm** [comment]
You can go ahead and make a PR for these changes and I can try and get the code changes to get them in once I can see how it will look. If I go into that code again, it may be an opportunity to make it so that the reports run on the version of the GO data that exists during the pipeline run, rather than the last release.

**Comment by ValWood** [comment]
It would be useful if they appear in the normal logs. Otherwise people will never know about them and fix them?

**Comment by ValWood** [comment]
PR?

**Comment by kltm** [comment]
Agreed, but that has not been a given priority as it would take a little work to merge them into the current reporting system--it is a set of "global" checks that are being made, rather than the line-by-line of all other currently implemented checks. This goes back to the history of them being completely separate systems, one being the general GAF checking system and the other a piece of bespoke code for examining annotation intersections in bulk.

**Comment by kltm** [comment]
A PR is a pull request in the GitHub repository. If you don't know how to make those, you can either email me or paste the file that you want into an appropriate ticket.

**Comment by ValWood** [comment]
I notices the other day when I was looking at the MAtrix  reports (based on @tberardini s questions), that the 
PANTHER family filtering is not yet in place (i.e violations are reported when they should not be).

I  had prepared a new rule set back in June. Is the stalling of this ticket because I did not do something? or did the filter not get added yet?)

Cheers

Val

@cmungall

**Comment by kltm** [comment]
@ValWood 
We are basically hung up on https://github.com/geneontology/go-site/issues/1110#issuecomment-505175511 . 

To get these into the usual GO Rules/reports location would take folding intersection code into a different code base, which would have to be specced out and "budgeted". (@pgaudet)
Otherwise, we could modify the current software to add the PANTHER column (https://github.com/geneontology/go-site/issues/1110#issuecomment-505169892) and add the new information needed to the metadata.

**Comment by ValWood** [comment]
Hi @kltm 
apologies this tickets was a conflation of 2 issues. 1 the separate logs, this is reported in other tickets I think? It is frustrating, but we can work around it until resources are available. It would be great if all of the checks were  organism specific, and in one place.

The main issue for this ticket, I believe (hope) is a  much easier  fix.

I create some exceptions based on family membership, if one orthologous group should be exclusively be allowed in an intersection. There are quite a lot of these, and they massively over-inflate the errors, will be confusing for curators (because the annotations are correct), and  basically if we can't fix this the checks will become impossible to check and correct.

It seems from the earlier comments above that I just needed to add these IDs to the relevant column which I believe I did (but let me know if you do not see them). Then I presumed you needed to make a small change to the filtering so that these were not reported in the logs.

**Comment by ValWood** [comment]
I have hidden the comments relating to the separate reporting

**Comment by ValWood** [comment]
Actually I just checked the file above,. There are no PANTHER IDs in there. 
Remind me where I should upload a new version of this file including the PATHER IDs

**Comment by kltm** [comment]
Essentially, to make this work the way I think you're describing, I there would be two steps:

- [ ] First, create a new metadata file (derived from our current https://github.com/geneontology/shared-annotation-check/blob/master/rules.txt) with the new fields added, to use as a reference for the next item
- [ ] Second, try and modify the existing script to operate on and filter with the new data above

The second item there is probably where most of the work would be (besides, of course, adding the families to the rules). Unfortunately, I have not been in that code in ages and it might take a little to orient and get it the way you needed; but from my description above, I seemed to have believed that it wouldn't be too hard. (That and finding a moment to crack it all open again.)
